src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -27,7 +27,20 @@ for dir in (:west, :east, :south, :north, :bottom, :top)
         @inline $extract_bc(bc::Tuple) = $extract_bc.(bc)
     end
 end
-
+# For inhomogeneous BC we extract the _last_ one 
+# example 
+# `bc.west <: DCBC`
+# `bc.east <: PBC`
+# `extract_west_or_east_bc(bc) == bc.west`
+    
+  extract_west_or_east_bc(bc) = sort([bc.west,   bc.east],  lt=fill_first)[2]
+extract_south_or_north_bc(bc) = sort([bc.south,  bc.north], lt=fill_first)[2]
+ extract_bottom_or_top_bc(bc) = sort([bc.bottom, bc.top],   lt=fill_first)[2]
+
+  extract_west_or_east_bc(bc::Tuple) =   extract_west_or_east_bc.(bc)
+extract_south_or_north_bc(bc::Tuple) = extract_south_or_north_bc.(bc)
+ extract_bottom_or_top_bc(bc::Tuple) =  extract_bottom_or_top_bc.(bc)
```

Comment:
Remember that often for small things it can be better to use `map` than broadcasting (which allocates more). But it depends on what you're doing.